core/spec/models/spree/shipment_spec.rb

```diff
@@ -25,64 +25,42 @@
   let(:variant) { mock_model(Spree::Variant) }
   let(:line_item) { mock_model(Spree::LineItem, variant: variant) }
 
-  # Regression test for #4063
-  context "number generation" do
-    before do
-      allow(order).to receive :update!
-    end
-
-    it "generates a number containing a letter + 11 numbers" do
-      shipment.save
-      expect(shipment.number[0]).to eq("H")
-      expect(/\d{11}/.match(shipment.number)).not_to be_nil
-      expect(shipment.number.length).to eq(12)
-    end
-  end
-
-  it 'is backordered if one if its inventory_units is backordered' do
-    shipment.inventory_units = [
-      build(:inventory_unit, state: 'backordered', shipment: nil),
-      build(:inventory_unit, state: 'shipped', shipment: nil),
-    ]
-    expect(shipment).to be_backordered
-  end
-
-  context '#determine_state' do
-    it 'returns canceled if order is canceled?' do
+  describe "#determine_state" do
+    it "returns canceled if order is canceled?" do
       allow(order).to receive_messages canceled?: true
-      expect(shipment.determine_state(order)).to eq 'canceled'
+      expect(shipment.determine_state(order)).to eq "canceled"
```

Comment:
If possible I'd like to see these `shipment.determine_state(order)`  expressions pulled out into a subject block in this pr